types/tx.go

```diff
@@ -41,3 +45,19 @@ type TxProof struct {
 	Data     Tx               `json:"data"`
 	Proof    merkle.Proof     `json:"proof"`
 }
+
+func (txs Txs) ToTxsWithISRs(intermediateStateRoots IntermediateStateRoots) ([]pb.TxWithISRs, error) {
+	expectedLength := len(txs) + 2
```

Comment:
Shouldn't this be + 1? 
Tx1 Tx2 -> ISR1 Tx1 ISR2 Txs ISR3

Reply:
No, it should be +2 because of `BeginBlock` and `EndBlock`. See https://github.com/rollkit/rollkit/blob/bbdc6cdc6430589a5ba2f14ccbfe42323a65750c/state/executor.go#L324

Reply:
ahh ok. So we have BeginBlock EndBlock, which are not Transactions and let`s say Tx1 Tx2.

BeginBlock Tx1 Tx2 Endblock -> [ISR0,ISR1] [ISR1, Tx1, ISR2] [ISR2, Tx2, ISR3] [ISR3,ISR4]

In this case, we have 5 ISRs for 2 Transactions.

Reply:
Yeah I understand but for `BeginBlock`, only the post ISR is in the list so there is no ISR0, there is only ISR1. This method only handles `Txs` so it'll output [ISR1, Tx1, ISR2] [ISR2, Tx2, ISR3] given Tx1 and Tx2.

Reply:
We should:
- generateISR before BeginBlock
- Put in begin / End Block as empty [] Txs

Reply:
Covered putting begin and endblock as empty [] Txs. Will add `generateISR` before BeginBlock in a separate PR.